docs/contribute/dev-contribute.mdx

```diff
@@ -30,27 +30,34 @@ cd prefect
 you can install [an editable version](https://setuptools.pypa.io/en/latest/userguide/development_mode.html) of Prefect for quick iteration.
 We recommend using a virtual environment such as `venv`:
 
-```bash
+<CodeGroup>
+```bash using pip and venv
 python -m venv .venv
 source .venv/bin/activate
 
 # Installs the package with development dependencies
 pip install -e ".[dev]"
 ```
 
+```bash using uv
+uv venv --python 3.12
+source .venv/bin/activate
+
+# Installs the package with development dependencies
+uv pip install -e ".[dev]" -U --force-reinstall
+```
+</CodeGroup>
+
 To ensure your changes comply with our linting policies, you can optionally set up pre-commit hooks to run with every commit:
 
 ```
 pre-commit install
 ```
 
-If you don't want to install the pre-commit hooks, you can manually install the linting utilities and run them yourself:
```

Comment:
i don't see why we'd want to offer this flexibility given that the pre-commits run `ruff` in a specifically different way that gives the impression of things being out of sync like in the motivating issue

Reply:
Yeah, this is what we do in CI too, so this makes sense to me.